internal/scaffold/cmd.go

```diff
@@ -162,15 +163,15 @@ func main() {
 	}
 	// Create Service object to expose the metrics port(s).
 	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
-	if err != nil {
+	if err != nil && !errors.Is(err, k8sutil.ErrRunLocal) {
 		log.Info("Could not create metrics Service", "error", err.Error())
 	}
 
     // CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
     // necessary to configure Prometheus to scrape metrics from this operator.
     services := []*v1.Service{service}
     _, err = metrics.CreateServiceMonitors(cfg, namespace, services)
-    if err != nil {
+    if err != nil && !errors.Is(err, k8sutil.ErrRunLocal) {
```

Comment:
[`CreateServiceMonitors`](https://github.com/operator-framework/operator-sdk/blob/master/pkg/metrics/service-monitor.go#L35) does not return `ErrRunLocal`.